cluster/evaluator/launch_eval.py

```diff
@@ -71,15 +75,46 @@ def same_run_eval(black_num=0, white_num=0):
 
     b = fsdb.get_model(black_num)
     w = fsdb.get_model(white_num)
-    bucket = fsdb.eval_dir
 
     b_model_path = os.path.join(fsdb.models_dir(), b)
     w_model_path = os.path.join(fsdb.models_dir(), w)
 
     launch_eval_job(b_model_path + ".pb",
                w_model_path + ".pb",
                "{:d}-{:d}".format(black_num, white_num),
-               bucket)
+               flags.FLAGS.bucket_name)
+
+
+def add_uncertain_pairs(dry_run=False):
+    new_pairs = ratings.suggest_pairs()
+    desired_pairs = restore_pairs() or []
+    desired_pairs += new_pairs
+    print("added %d new pairs" % len(new_pairs))
+    print(set([p[0] for p in new_pairs]))
+    for p in new_pairs:
+        print(p)
+    if not dry_run:
+        with open('pairlist.json', 'a') as f:
+            json.dump(new_pairs, f)
+        save_pairs(desired_pairs)
+
+def add_top_pairs(dry_run=False):
+    top = ratings.top_n(20)
+    new_pairs = []
+    for idx, t in enumerate(top[:15]):
+      new_pairs += [[t[0], o[0]] for o in top[idx+1:idx+5]]
+    print(new_pairs)
+    desired_pairs = restore_pairs() or []
+    desired_pairs += new_pairs
+    print("added %d new pairs" % len(new_pairs))
+    print(set([p[0] for p in new_pairs]))
+    for p in new_pairs:
+        print(p)
+    if not dry_run:
+        with open('pairlist.json', 'a') as f:
```

Comment:
this appears to be overridden by save_pairs

Reply:
Refactored.